samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -188,6 +195,10 @@ void doOnProcessorChange(List<String> processors) {
 
     // Generate the JobModel
     JobModel jobModel = generateNewJobModel(currentProcessorIds);
+    if (!hasCreatedChangeLogStreams) {
```

Comment:
instead of boolean we should check number of partitions. Start with value 0. If number of partitions matches - skip creation.